### PR TITLE
ayatana-indicator-display: init & add to Lomiri

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

Inits [ayatana-indicator-display](https://github.com/AyatanaIndicators/ayatana-indicator-display) and adds it to the Lomiri module.

On Lomiri, it only handles screen lock afaict. Not much to show there.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by OPNA2608** [comment]
@ofborg build nixosTests.lomiri

File: nixos/tests/ayatana-indicators.nix [hunk 1]
```diff
@@ -39,6 +40,7 @@ in {
     # Setup needed by some indicators
 
     services.accounts-daemon.enable = true; # messages
+    # services.geoclue2.enable = true; # display, maybe not a good idea to enable this for reproducibility?
```
Comment (SuperSandro2000) [inline nixos/tests/ayatana-indicators.nix:43]: Do you remember why this is then not reproducible?

**Review by SuperSandro2000 — COMMENTED** [review]
Please rebase

File: nixos/tests/ayatana-indicators.nix (hunk 1 above)
Comment (OPNA2608) [reply nixos/tests/ayatana-indicators.nix:43]: I haven't tested this, I just didn't think that tests should have any geolocation interactions happening inside of them. AI display can use `xsct` to change the screen temperature based on geoclue information, not sure if that could i.e. mess up any of the OCR.

**Comment by OPNA2608** [comment]
Bumped to new release.

File: nixos/tests/ayatana-indicators.nix (hunk 1 above)
Comment (SuperSandro2000) [reply nixos/tests/ayatana-indicators.nix:44]: If the screen temperature gets to red, then the lower contrast could mess with things. I would say we just drop this comment to not confuse us in the future.

Comment (OPNA2608) [reply nixos/tests/ayatana-indicators.nix]: Dropped.

**Comment by OPNA2608** [comment]
Ugh, tests fail because `lomiri.telephony-service` is failing again. Builds fine locally, so maybe more flakiness:

https://github.com/NixOS/nixpkgs/blob/2db5894f3504741b1ca909b46a1afee29bd412c3/pkgs/desktops/lomiri/services/telephony-service/default.nix#L130-L140

@ofborg build ayatana-indicator-display ayatana-indicator-display.passthru.tests

**Comment by OPNA2608** [comment]
And now a test in `lomiri.lomiri-indicator-network` fails with an identical issue:
```
[ RUN      ] TestSecretAgent.MultiSecrets
1716390968.777 GetServerInformation
1716390968.778 RegisterWithCapabilities "com.lomiri.indicator.network.SecretAgent" 0
"/connection/foo" "802-11-wireless-security" () 5
Requesting secret from user
1716390968.782 Notify "lomiri-indicator-network" 0 "wifi-full-secure" "Connect to “the ssid”" "WPA" ["connect_id", "Connect", "cancel_id", "Cancel"] {"x-lomiri-non-shaped-icon": "true", "x-lomiri-private-affirmative-tint": "true", "x-lomiri-private-menu-model": {"actions": {"notifications": "/action/1"}, "busName": ":1.3", "menuPath": "/menu/1"}, "x-lomiri-snap-decisions": "true", "x-lomiri-snap-decisions-timeout": 2147483647} 0
"/connection/foo2" "802-11-wireless-security" () 5
Requesting secret from user
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: invalid (NULL) pointer instance
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: invalid (NULL) pointer instance
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: invalid (NULL) pointer instance
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: invalid (NULL) pointer instance
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
(process:2428): GLib-CRITICAL **: 15:16:08.782: g_hash_table_unref: assertion 'hash_table != NULL' failed
(process:2428): GLib-CRITICAL **: 15:16:08.782: g_main_context_unref: assertion 'context != NULL' failed
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
(process:2428): GLib-GObject-CRITICAL **: 15:16:08.782: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
free(): double free detected in tcache 2
```

Both started failing on Hydra today: [`lomiri.telephony-service`](https://hydra.nixos.org/build/260540921), [`lomiri.lomiri-indicator-network`](https://hydra.nixos.org/build/260438136). Maybe an issue with the latest staging-next merge, I'll look into it tonight. :slightly_frowning_face:

FWIW, `ayatana-indicator-display` itself works fine.

**Comment by OPNA2608** [comment]
Failures in other packages & ayatana-indicators test will be fixed by https://github.com/NixOS/nixpkgs/pull/314043, this is a bug in GLib.

Will rebase ASAP.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**